CSharp/README.md

```diff
@@ -23,6 +24,115 @@ var httpClient = new HttpClient();
 IUserService userService = new UserService();
 ```
 
+### Don't use explicit *this* reference
+
+It is redundant and does not add for readability. For more information read [this SO post](http://stackoverflow.com/questions/5885249/to-use-or-not-to-use-the-this-qualifier-in-c-sharp).
+
+```csharp
+// Bad
+this.ValidateParameters();
+
+// Good
+ValidateParameters();
+```
+
+### Use *string* instead of *String* keyword
+
+```csharp
+// Bad
+String.IsNullOrEmpty(name);
+
+// Good
+string.IsNullOrEmpty(name);
+```
+
+### Use readonly fields when possible
+
+### Use string interpolation
+
+```csharp
+// Bad
+var message = string.Format("My name is {0} {1}.", person.FirstName, person.LastName);
+
+// Good
+var message = $"My name is {person.FirstName} {person.LastName}.");
+```
+
+## Layout
+
+### Use *four spaces* per indentation level
+
+```csharp
+// Bad
+public void SomeMethod()
+{
+  DoSomethingFirst();
+  DoSomerthingLater();
+}
+
+// Good
+public void SomeMethod()
+{
+    DoSomethingFirst();

```

Comment:
The JavaScript code style had a clever way of showing this: it used a particular character to display the spaces.

Do you think it would be a good idea to show that here too?